pypga/modules/migen/spimachine.py

```diff
@@ -33,6 +33,7 @@ def __init__(self, width):
                     If(do_extend,
                         do_extend.eq(0)
                     ).Else(
+                        #
                         cnt.eq(self.div[1:]),
                         do_extend.eq(self.extend & self.div[0])
```

Comment:
I understand that ClockGen is basically a counter that controls how many instruction cycles of the FPGA one SPI clock half cycle lasts. And it seems that it's convenient to have this `extend` feature. But why does this look at the LSB of `div` and then set `cnt` to the higher bits of `div`? Do you have any idea?